tests/hazmat/primitives/test_pkcs12.py

```diff
@@ -137,3 +138,55 @@ def test_buffer_protocol(self, backend):
         assert parsed_key is not None
         assert parsed_cert is not None
         assert parsed_more_certs == []
+
+    @pytest.fixture(scope="function")
+    def cert(self, backend):
+        return load_vectors_from_file(
+            os.path.join("x509", "custom", "ca", "ca.pem"),
+            lambda pemfile: x509.load_pem_x509_certificate(
+                pemfile.read(), backend
+            ), mode="rb"
+        )
+
+    @pytest.fixture(scope="function")
+    def key(self, backend):
+        return load_vectors_from_file(
+            os.path.join("x509", "custom", "ca", "ca_key.pem"),
+            lambda pemfile: load_pem_private_key(
+                pemfile.read(), None, backend
+            ), mode="rb"
+        )
+
+    @pytest.mark.parametrize("key_encryption_algorithm", [
+        None,
+        serialization.BestAvailableEncryption(b"pass123"),
+        serialization.NoEncryption()
+    ])
+    @pytest.mark.parametrize("cert_encryption_algorithm", [
+        None,
+        serialization.BestAvailableCertificateEncryption(),
+        serialization.NoCertificateEncryption(),
+        # NID_pbe_WithSHA1And40BitRC2_CBC
+        serialization.CustomCertificateEncryption(149)
+    ])
+    def test_generate(self, backend, key, cert, key_encryption_algorithm,
+                      cert_encryption_algorithm):
+        password = b"cryptography"
+        p12 = generate_pkcs12(
+            password, None, key, cert, None, backend)
+        parsed_key, parsed_cert, parsed_more_certs = \
+            load_key_and_certificates(p12, password, backend)
+        assert parsed_cert == cert
+        assert parsed_key.private_numbers() == key.private_numbers()
+        assert parsed_more_certs == []
+
+    def test_generate_bad(self, backend, key, cert):
+        if 'NID_X25519' in dir(backend._lib):
```

Comment:
this isn't the right way to check if a backend support x25519.

Reply:
Can you please suggest a way to do it?

Reply:
https://github.com/pyca/cryptography/blob/master/tests/hazmat/primitives/test_x25519.py#L23